src/java/htsjdk/samtools/cram/structure/Slice.java

```diff
@@ -66,6 +68,7 @@
     public SAMBinaryTagAndValue sliceTags;
 
     private void alignmentBordersSanityCheck(final byte[] ref) {
+        if (sequenceId == UNMAPPED_OR_NO_REFERENCE) return ;
         if (alignmentStart > 0 && sequenceId >= 0 && ref == null) throw new NullPointerException("Mapped slice reference is null.");
```

Comment:
This seems like it should be either IllegalArgumentException or IllegalStateException.